extensions/azurecore/src/azureResource/interfaces.ts

```diff
@@ -40,3 +40,9 @@ export interface IAzureResourceNodeWithProviderId {
 export interface IAzureResourceService<T extends azureResource.AzureResource> {
 	getResources(subscriptions: azureResource.AzureResourceSubscription[], credential: msRest.ServiceClientCredentials, account: Account): Promise<T[]>;
 }
+
+export class ObjectExplorerActionsContext implements ObjectExplorerContext {
```

Comment:
I still don't see why you need this class (especially now that you aren't using instanceof). You should be able to remove it and not change any of your other logic.

Reply:
Because I don't want to use `any` for the node type. If we want to use the properties, we need to get this class. The ExtHostOEShape converts this onto this class.

Reply:
You can just use azdata.ObjectExplorerContext. That has the properties already.

Reply:
it's an interface

Reply:
And? Interfaces have properties. You're not trying to actually create an instance of the class or anything.

Reply:
The node type needs to be a class. I'd rather not commit with compilation warnings

Reply:
The other extensions just cast it or send the node as an object. This was to provide functionality to work with OE I believe

Reply:
Nevermind, I saw some errors earlier, but you're right, we don't need it